File: FD/C#/ASVLOFFSCREEN.cs
using System;
using System.Runtime.InteropServices;

namespace arcsoft
{

    [StructLayout(LayoutKind.Sequential)]
    public struct ASVLOFFSCREEN
    {
        public int u32PixelArrayFormat;
        public int i32Width;
        public int i32Height;

        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 4)]
        public IntPtr[] ppu8Plane;

        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 4)]
        public int[] pi32Pitch;

        public void freeUnmanaged() {
            if (ppu8Plane != null) {
                for (int i = 0; i < ppu8Plane.Length; i++) {
                    if (ppu8Plane[i] != IntPtr.Zero) {
                        Marshal.FreeHGlobal(ppu8Plane[i]);
                    }
                }
                ppu8Plane = null;
            }
        }
    }
}

File: FR/java/src/com/arcsoft/ASVLOFFSCREEN.java
package com.arcsoft;

import java.util.Arrays;
import java.util.List;

import com.sun.jna.Pointer;
import com.sun.jna.Structure;

public class ASVLOFFSCREEN extends Structure {
    public int u32PixelArrayFormat;
    public int i32Width;
    public int i32Height;
    public Pointer[] ppu8Plane = new Pointer[4];
    public int[] pi32Pitch = new int[4];
    
    public ASVLOFFSCREEN(){
    
    }

    @Override
    protected List getFieldOrder() { 
        return Arrays.asList(new String[] { 
            "u32PixelArrayFormat", "i32Width", "i32Height", "ppu8Plane","pi32Pitch"
        });
    }
}
